test/parser.test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { parseDataUnit, parseDataUnitType } = require('../lib/parser');
+const {
+  ENCODINGS,
+  COMPRESSIONS,
+  HANDSHAKE_STATUSES,
+  DATA_UNIT_TYPES,
+} = require('../lib/constants');
+
+const {
+  generateRandomInt,
+  generateRandomUInt16,
+  generateRandomUInt32,
+} = require('./utils');
+
+test('should parse protocol data unit type', () => {
+  const typeId = generateRandomInt(0, 4);
+  const type = DATA_UNIT_TYPES[typeId];
+  const buffer = Buffer.from([3, 4, typeId, 1, 2]);
+  const parsedType = parseDataUnitType(buffer, 2);
+  expect(parsedType).toStrictEqual(type);
+});
+
+test('should parse handshake', () => {
+  const statusId = generateRandomInt(0, HANDSHAKE_STATUSES.length - 1);
```

Comment:
```suggestion
  const statusId = generateRandomInt(0, HANDSHAKE_STATUSES.length);
```

Reply:
https://github.com/crewaii/protocol/blob/ce01d039042f0e6f7a0a8680acb2fd1c86028e99/test/utils.js#L3
```generateRandomInt``` genarates random number in [min, max). Max value of statusId is  HANDSHAKE_STATUSES.length - 1;